crates/router/src/core/errors/api_error_response.rs

```diff
@@ -341,6 +343,7 @@ impl common_utils::errors::ErrorSwitch<api_models::errors::types::ApiErrorRespon
                 ApiError::new("IR", 21, "Missing required params".to_string(), Some(Extra {data: Some(serde_json::json!(field_names)), ..Default::default() })),
             ),
             Self::AccessForbidden => AER::ForbiddenCommonResource(ApiError::new("IR", 22, "Access forbidden. Not authorized to access this resource", None)),
+            Self::UnprocessableEntity => AER::Unprocessable(ApiError::new("IR", 23, "Token expired or invalid", None)),
```

Comment:
You can consider making it more generic instead of hardcoding to Token expired because this error could be used else where other than token flow.